### PR TITLE
Add invite to organization

### PR DESCRIPTION
**Pull request by bwhicks**
This requests exposes the API functionality for inviting a user to an organization, including a flag to set them as admin ("owner") on invite.

It includes the new method for Organization, unit tests and testing data fixtures.

### CODE REVIEW

**Comment by babineaum** [comment]
👍

**Comment by blckjack** [comment]
@bwhicks  pulled your changes, but it seems that has_in_members method doesn't work, please assist!

**Comment by bwhicks** [comment]
@blckjack Could you clarify what's going wrong with `Organization.has_in_members()`? I can't see obviously how my change could break that functionality, but that certainly doesn't mean something weird didn't go on.

**Comment by cruftyoldsysadmin** [comment]
Will this be merged anytime soon?

File: github/Organization.py [hunk 1]
```diff
@@ -596,6 +597,31 @@ def has_in_public_members(self, public_member):
         )
         return status == 204
 
+    def invite_to_organization(self, user, admin=None):
+        """
+        :calls: `PUT /orgs/:org/memberships/:username`_
```
Comment (sfdye) [inline github/Organization.py]: Would you mind adding the GitHub link to this endpoint?

File: github/Organization.py [hunk 2]
```diff
@@ -596,6 +597,31 @@ def has_in_public_members(self, public_member):
         )
         return status == 204
 
+    def invite_to_organization(self, user, admin=None):
```
Comment (sfdye) [inline github/Organization.py]: Can you add `role` has a parameter and assert the values of role to only accept "admin" or "member", and default to "member" if nothing is provided. Also this method does not need to return anything. You can refer the `Team.add_membership()` as an example.

File: github/Organization.py (hunk 1 above)
Comment (bwhicks) [reply github/Organization.py]: Not a problem. Sorry I missed it.

Comment (sfdye) [reply github/Organization.py]: Let's get this merged soon 😄

**Comment by Constantin07** [comment]
Any idea when this PR will be merged ?

**Comment by bwhicks** [comment]
@Constantin07 I hope soon. I need some time to refactor per @sfdye 's code review, and being able to test that meaningfully will imply adding one other bit of functionality from invites. Unfortunately, this sat for roughly nine months before the code review hit, and I'm much busier now than when I originally wrote it. Alas.

**Comment by bwhicks** [comment]
@Constantin07 I hope soon, but @sfdye requested changes which will take both a refactor and probably some additional functionality for tests that actually meaningfully test that I am inviting people to the organization. Unfortunately, this sat for roughly 9 months without feedback and I am much busier now than when I wrote the original PR.

**Comment by bwhicks** [comment]
I am closing this PR in favor of #692 using `NamedUser`. Testing it properly will require adding `Membership` given the pattern requested, and I simply do not have time now. Sorry.

**Comment by Constantin07** [comment]
Thanks @bwhicks.

**Pull request closed without merging**